src/psyclone/tests/psyir/symbols/symbol_test.py

```diff
@@ -65,9 +74,146 @@ def test_symbol_initialisation():
     assert ("Symbol 'visibility' attribute should be of type "
             "psyir.symbols.Symbol.Visibility but" in str(error.value))
 
+    with pytest.raises(TypeError) as error:
+        Symbol('sym1', interface="hello")
+    assert ("The interface to a Symbol must be a SymbolInterface but got "
+            "'str'" in str(error.value))
+
+
+def test_symbol_interface_setter():
+    '''Test that the Symbol interface setter behaves as expected,
+    including raising an exception if the input is of the wrong
+    type. Also use this to test the is_local, is_global and
+    is_argument and is_unresolved properties.
+
+    '''
+    symbol = Symbol('sym1')
+    assert symbol.is_local
+    assert not symbol.is_global
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+
+    symbol.interface = GlobalInterface(ContainerSymbol("my_mod"))
+    assert not symbol.is_local
+    assert symbol.is_global
+    assert not symbol.is_argument
+    assert not symbol.is_unresolved
+
+    symbol.interface = ArgumentInterface()
+    assert not symbol.is_local
+    assert not symbol.is_global
+    assert symbol.is_argument
+    assert not symbol.is_unresolved
+
+    symbol.interface = UnresolvedInterface()
+    assert not symbol.is_local
+    assert not symbol.is_global
+    assert not symbol.is_argument
+    assert symbol.is_unresolved
+
+    with pytest.raises(TypeError) as info:
+        symbol.interface = "hello"
+    assert ("The interface to a Symbol must be a SymbolInterface but got "
+            "'str'" in str(info.value))
+
 
 def test_symbol_str():
     '''Test that a Symbol instance can be stringified'''
 
     sym = Symbol("my_symbol")
     assert str(sym) == "my_symbol"
+
+
+# SymbolError is declared but not used in symbol.py. There are many
```

Comment:
Bring this comment to the top, or maybe even inside the module docstring?

Reply:
Yes, good idea.